app/js/services/situationService.js

```diff
@@ -197,6 +197,32 @@ angular.module('ddsApp').factory('SituationService', function($http, $sessionSto
 
         hasEnfant: function(situation) {
             return _.find(situation.individus, { role: 'enfant' });
+        },
+
+        hasConjoint: function(situation) {
+            return _.find(situation.individus, { role: 'conjoint' });
+        },
+
+        setConjoint: function(situation, conjoint) {
+            var individus = situation.individus;
+            // si le conjoint existait déjà avant, on l'écrase
+            if (this.hasConjoint(situation)) {
+                individus[individus.length - 1] = conjoint;
```

Comment:
This is probably not worth changing if it was already the case before and you just encapsulated the logic, but this is extremely fragile. It makes the implicit assumption that `situation.individus` is private and will only ever be changed through `setConjoint`, otherwise calling `setConjoint` would overwrite existing data.